### PR TITLE
envoy: build with full Xcode

### PR DESCRIPTION
**Pull request by codefromthecrypt**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/envoy.rb [hunk 1]
```diff
@@ -7,9 +7,9 @@ class Envoy < Formula
       tag:      "v1.20.1",
       revision: "ea23f47b27464794980c05ab290a3b73d801405e"
   license "Apache-2.0"
+  # Ex. brew install --build-from-source envoy --HEAD
```
Comment (carlocab) [inline Formula/envoy.rb:10]:
```suggestion
```

File: Formula/envoy.rb [hunk 2]
```diff
@@ -24,6 +24,9 @@ class Envoy < Formula
   depends_on "libtool" => :build
   depends_on "ninja" => :build
   depends_on macos: :catalina
+  # Starting with 1.21, envoy requires a full Xcode installation, not just
+  # command-line tools. See envoyproxy/envoy#16482
+  depends_on xcode: :build
```
Comment (carlocab) [inline Formula/envoy.rb:29]:
```suggestion
  # Starting with 1.21, envoy requires a full Xcode installation, not just
  # command-line tools. See envoyproxy/envoy#16482
  depends_on xcode: :build
  depends_on macos: :catalina
```

Comment (carlocab) [inline Formula/envoy.rb:29]:
```suggestion
  # Starting with 1.21, envoy requires a full Xcode installation, not just
  # command-line tools. See envoyproxy/envoy#16482
  depends_on xcode: :build
  depends_on macos: :catalina
```

**Review by carlocab — COMMENTED** [review]
Thanks, @codefromthecrypt. Could you squash these suggestions into your first commit, and reword it to
```
envoy: build with full Xcode
```
or similar? We put the names of formulae first in our commit messages.

File: Formula/envoy.rb (hunk 2 above)
Comment (carlocab) [reply Formula/envoy.rb:28]: Oops, not sure why this is showing up twice.

**Comment by carlocab** [comment]
Oh, wait, your commit message already does that. I was misled by the PR title. My mistake.

**Comment by codefromthecrypt** [comment]
it's ok, needed to squash anyway, after reading my title I changed the commit, but forgot to change the title. sorry about the confusion

**Review by carlocab — APPROVED** [review]
Any reason this is still a draft?

**Comment by codefromthecrypt** [comment]
> Any reason this is still a draft?

it takes time for the build to complete so I was waiting until they finished. I noticed just now the same error as before on my Mac silicon.

```
$ brew install --build-from-source envoy --HEAD
 
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/core).
==> Updated Formulae
Updated 65 formulae.
==> Deleted Formulae
sdhash

==> Cloning https://github.com/envoyproxy/envoy.git
Updating /Users/thebots/Library/Caches/Homebrew/envoy--git
From https://github.com/envoyproxy/envoy
   f4bdebe68..8ea1ef2f3  main       -> origin/main
==> Checking out branch main
Previous HEAD position was 8a9779a61 build(deps): bump setuptools from 59.0.1 to 59.5.0 in /tools/base (#19201)
Switched to a new branch 'main'
Branch 'main' set up to track remote branch 'main' from 'origin'.
HEAD is now at 8ea1ef2f3 remove unnecessary hash searching of default http upstream conn pool factory (#19104)
==> /opt/homebrew/opt/bazelisk/bin/bazelisk build --compilation_mode=opt --curses=no --show_tas
Last 15 lines from /Users/thebots/Library/Logs/Homebrew/envoy/01.bazelisk:
Analyzing: target //source/exe:envoy-static (20 packages loaded, 7 targets configured)
Analyzing: target //source/exe:envoy-static (20 packages loaded, 7 targets configured)
Analyzing: target //source/exe:envoy-static (20 packages loaded, 7 targets configured)
Analyzing: target //source/exe:envoy-static (20 packages loaded, 7 targets configured)
Analyzing: target //source/exe:envoy-static (20 packages loaded, 7 targets configured)
Analyzing: target //source/exe:envoy-static (20 packages loaded, 7 targets configured)
ERROR: /private/tmp/envoy-20211208-67008-fwq64c/.brew_home/_bazel/38c2715f563f19e2836a01916f2a2c39/external/local_config_cc/BUILD:48:19: in cc_toolchain_suite rule @local_config_cc//:toolchain: cc_toolchain_suite '@local_config_cc//:toolchain' does not contain a toolchain for cpu 'darwin_arm64'
ERROR: While resolving configuration keys for //bazel:gcc_build: Analysis of target '@local_config_cc//:toolchain' failed
ERROR: While resolving configuration keys for //source/exe:envoy-static: Analysis of target '@local_config_cc//:toolchain' failed
ERROR: Analysis of target '//source/exe:envoy-static' failed; build aborted: Analysis of target '@local_config_cc//:toolchain' failed
INFO: Elapsed time: 66.581s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (34 packages loaded, 202 targets configured)
FAILED: Build did NOT complete successfully (34 packages loaded, 202 targets configured)

READ THIS: https://docs.brew.sh/Troubleshooting

These open issues may also help:
envoy: build with full Xcode https://github.com/Homebrew/homebrew-core/pull/90695
Please create pull requests instead of asking for help on Homebrew's GitHub,
Twitter or any other official channels.
```

I guess homebrew doesn't enforce the Xcode setup? If not, I'd like to add a comment with the command to switch it. sound ok?

**Comment by codefromthecrypt** [comment]
FYI on my arm laptop, full Xcode wasn't installed, but command line tools were. I'm currently manually installing Xcode (not sure if homebrew could manage it or not). It would have been ideal to quit with an error to install Xcode given what's in the ruby file.

```
$ xcode-select --print-path
/Library/Developer/CommandLineTools
```

**Comment by codefromthecrypt** [comment]
FYI full Xcode seems like it will take hours to download, so I don't expect to report back any news on this actually working today.

Regardless, if you don't mind, I'd like to add the comment to do `sudo xcode-select --switch /Applications/Xcode.app` as others had to discover this in another issue and saying it explicitly seems warranted until that is automated.

**Comment by carlocab** [comment]
> Regardless, if you don't mind, I'd like to add the comment to do `sudo xcode-select --switch /Applications/Xcode.app` as others had to discover this in another issue and saying it explicitly seems warranted until that is automated.

This shouldn't be needed as long as the formula has `depends_on xcode: :build`.

**Comment by codefromthecrypt** [comment]
agree you are right. something must have reverted my change when I tried to test this. It looks ok on the host missing Xcode (well that is still downloading it):

```
$ brew install --build-from-source envoy --HEAD
envoy: A full installation of Xcode.app is required to compile
this software. Installing just the Command Line Tools is not sufficient.

Xcode can be installed from the App Store.
Error: envoy: An unsatisfied requirement failed this build.
```
